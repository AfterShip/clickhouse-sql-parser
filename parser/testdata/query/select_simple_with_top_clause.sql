SELECT TOP 10 my_column FROM tableName;
