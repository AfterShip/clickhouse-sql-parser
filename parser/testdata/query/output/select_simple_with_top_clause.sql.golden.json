[
  {
    "SelectPos": 0,
    "StatementEnd": 38,
    "With": null,
    "Top": {
      "TopPos": 7,
      "TopEnd": 13,
      "Number": {
        "NumPos": 11,
        "NumEnd": 13,
        "Literal": "10",
        "Base": 10
      },
      "WithTies": false
    },
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "Name": "my_column",
          "QuoteType": 1,
          "NamePos": 14,
          "NameEnd": 23
        },
        "Modifiers": [],
        "Alias": null
      }
    ],
    "From": {
      "FromPos": 24,
      "Expr": {
        "Table": {
          "TablePos": 29,
          "TableEnd": 38,
          "Alias": null,
          "Expr": {
            "Database": null,
            "Table": {
              "Name": "tableName",
              "QuoteType": 1,
              "NamePos": 29,
              "NameEnd": 38
            }
          },
          "HasFinal": false
        },
        "StatementEnd": 38,
        "SampleRatio": null,
        "HasFinal": false
      }
    },
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": null,
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": null,
    "Except": null
  }
]